packages/scratch-gui/test/unit/components/icon-button.test.jsx

```diff
@@ -1,37 +1,56 @@
 import React from 'react';
-import {shallow} from 'enzyme';
+import { render, fireEvent } from '@testing-library/react';
 import IconButton from '../../../src/components/icon-button/icon-button';
-import renderer from 'react-test-renderer';
+import '@testing-library/jest-dom';
+
 
 describe('IconButtonComponent', () => {
-    test('matches snapshot', () => {
-        const onClick = jest.fn();
-        const title = <div>Text</div>;
-        const imgSrc = 'imgSrc';
-        const className = 'custom-class-name';
-        const component = renderer.create(
-            <IconButton
-                className={className}
-                img={imgSrc}
-                title={title}
-                onClick={onClick}
-            />
-        );
-        expect(component.toJSON()).toMatchSnapshot();
+    const defaultProps = {
+        img: 'imgSrc',
+        title: <div>Text</div>,
+        onClick: jest.fn(),
+        className: 'custom-class-name'
+    };
+
+    test('renders with all props correctly', () => {
+        const { container } = render(<IconButton {...defaultProps} />);
+        
+        const button = container.querySelector('div[role="button"]');
+        expect(button).toBeTruthy();
+        expect(button).toHaveClass('custom-class-name');
+        
+        const image = container.querySelector('img');
+        expect(image).toBeTruthy();
+        expect(image).toHaveAttribute('src', 'imgSrc');
+        expect(image).toHaveAttribute('draggable', 'false');
+        
+        const text = container.querySelector('div div div');
+        expect(text).toHaveTextContent('Text');
     });
 
-    test('triggers callback when clicked', () => {
-        const onClick = jest.fn();
-        const title = <div>Text</div>;
-        const imgSrc = 'imgSrc';
-        const componentShallowWrapper = shallow(
-            <IconButton
-                img={imgSrc}
-                title={title}
-                onClick={onClick}
+    test('triggers callback only once when clicked', () => {
+        const handleClick = jest.fn();
+        
+        const { container } = render(
+            <IconButton 
+                {...defaultProps} 
+                onClick={handleClick} 
             />
         );
-        componentShallowWrapper.simulate('click');
-        expect(onClick).toHaveBeenCalled();
+        
+        const button = container.querySelector('div[role="button"]');
+        
+        if (!button) {
+            throw new Error('Button not found');
+        }
+
+        fireEvent.click(button);
+        
+        expect(handleClick).toHaveBeenCalledTimes(1);
+    });
+
+    test('matches snapshot', () => {
```

Comment:
If the component matches the snapshot it would be the same as the result of `renders with all props correctly`

Reply:
I would rather leave the snapshot test rather than `renders with all props correctly`. The second one will not fail if we add new props to the component and we may miss updating the test. Whereas if the snapshot is old it will fail